salt/sdb/redis_sdb.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+'''
+Redis SDB module
+================
+
+This module allows access to Redis  using an ``sdb://`` URI.
+
+Like all SDB modules, the Redis module requires a configuration profile to
+be configured in either the minion or master configuration file. This profile
+requires very little. For example:
+
+.. code-block:: yaml
+
+    sdb_redis:
+      driver: redis
+      host: 127.0.0.1
+      port: 6379
+      password: pass
+      db: 1
+
+The ``driver`` refers to the Redis module, all other options are optional.
+For option details see: https://redis-py.readthedocs.io/en/latest/.
+
+'''
+from __future__ import absolute_import
+
+try:
+    import redis
+    HAS_REDIS = True
+except ImportError:
+    HAS_REDIS = False
+
+
+__func_alias__ = {
+    'set_': 'set'
+}
+__virtualname__ = 'redis'
+
+
+def __virtual__():
+    '''
+    Module virtual name.
+    '''
+    if not HAS_REDIS:
+        return (False, 'Please install python-redis to use this SDB module.')
+    return __virtualname__
+
+
+def set_(key, value, profile=None):
+    '''
+    Set a value into the Redis SDB.
+    '''
+    if not profile:
+        return False
+    redis_kwargs = profile.copy()
+    redis_kwargs.pop('driver')
+    redis_conn = redis.Redis(**redis_kwargs)
```

Comment:
Fyi, we pack the minion modules into sdb.

https://github.com/saltstack/salt/blob/develop/salt/loader.py#L892

So, you should just be able to do `__salt__['redis.set'](key, value, **redis_kwargs)`

Reply:
I tried on 2017.7 and it doesn't work like that. Did that change in 2018.3?

Either way, the Redis module only accepts `host`, `password`, `port`, and `db` as arguments, while there are many more available (and with the code I proposed we don't impose any restriction - some of them are very important!) - see https://redis-py.readthedocs.io/en/latest/#contents. What do you think?

Reply:
OK, those should also be added to the redis module, but we don't have to do that now, but this should still use the StrictRedis class

Reply:
Yes, good catch, should use `StrictRedis` indeed.

Reply:
Pushed https://github.com/saltstack/salt/pull/48239/commits/3f17add324361e177fc9eab8bcfb98cf91109e18 to solve this.